client/src/Try/Gist.purs

```diff
@@ -5,45 +5,78 @@ module Try.Gist
   , tryLoadFileFromGist
   ) where
 
--- | An abstract data type representing the data we get back from the GitHub API.
 import Prelude
 
-import Control.Monad.Cont.Trans (ContT(..))
+import Affjax (printError)
+import Affjax as AX
+import Affjax.RequestBody as AXRB
+import Affjax.ResponseFormat as AXRF
+import Affjax.StatusCode (StatusCode(..))
 import Control.Monad.Except.Trans (ExceptT(..))
+import Data.Argonaut.Core (Json, caseJsonObject, stringify, toString)
 import Data.Either (Either(..))
-import Effect (Effect)
-import Effect.Uncurried (EffectFn1, EffectFn3, EffectFn4, mkEffectFn1, runEffectFn3, runEffectFn4)
+import Data.Function.Uncurried (Fn2, runFn2)
+import Data.Maybe (Maybe(..))
+import Data.Nullable (Nullable, toMaybe)
+import Effect.Aff (Aff)
+import Foreign.Generic (encodeJSON)
+import Foreign.Object as Object
+import Unsafe.Coerce (unsafeCoerce)
 
 -- | An abstract data type representing the data we get back from the GitHub API.
 data GistInfo
 
-foreign import uploadGist_
-  :: EffectFn3 String
-               (EffectFn1 String Unit)
-               (EffectFn1 String Unit)
-               Unit
-
--- | A wrapper for `uploadGist` which uses `ContT`.
-uploadGist :: String -> ExceptT String (ContT Unit Effect) String
-uploadGist content = ExceptT (ContT \k -> runEffectFn3 uploadGist_ content (mkEffectFn1 (k <<< Right)) (mkEffectFn1 (k <<< Left)))
-
--- | Get a gist by its ID
-foreign import getGistById_
-  :: EffectFn3 String
-               (EffectFn1 GistInfo Unit)
-               (EffectFn1 String Unit)
-               Unit
-
--- | A wrapper for `getGistById` which uses `ContT`.
-getGistById :: String -> ExceptT String (ContT Unit Effect) GistInfo
-getGistById id_ = ExceptT (ContT \k -> runEffectFn3 getGistById_ id_ (mkEffectFn1 (k <<< Right)) (mkEffectFn1 (k <<< Left)))
-
-foreign import tryLoadFileFromGist_
-  :: EffectFn4 GistInfo
-               String
-               (EffectFn1 String Unit)
-               (EffectFn1 String Unit)
-               Unit
-
-tryLoadFileFromGist :: GistInfo -> String -> ExceptT String (ContT Unit Effect) String
-tryLoadFileFromGist gi filename = ExceptT (ContT \k -> runEffectFn4 tryLoadFileFromGist_ gi filename (mkEffectFn1 (k <<< Right)) (mkEffectFn1 (k <<< Left)))
+type FileName = String
+
+type RawURL = String
```

Comment:
I think we should avoid type synonyms like these: they give a false sense of security and I think they require you to keep more in your head while reading the code. In fact we came quite close to making type synonyms like these a warning in the compiler once: https://github.com/purescript/purescript/pull/2932.

If it's important enough to distinguish these types to stop you from putting one in where you meant the other, or for documentation purposes, then at least I think they should be newtypes. Otherwise I think we should just use `String`.

Reply:
I think this could easily go either way. I’ll think a bit more on this when I come back to it, but I’ll probably revert to String rather than add newtypes. I agree about the false sense of security.